File: ios/dataProvider/contracts/FileContract.m
//
//  FileContract.m
//  StorjMobile
//
//  Created by Barterio on 3/15/18.
//  Copyright © 2018 Storj. All rights reserved.
//

#import "FileContract.h"

@implementation FileContract

static NSString * const _TABLE_NAME = @"files";

//static NSString * const _ID = @"id";
static NSString * const _FILE_ID = @"fileId";
static NSString * const _NAME = @"name";
static NSString * const _MIME_TYPE = @"mimeType";
static NSString * const _INDEX = @"_index";
static NSString * const _HMAC = @"hmac";
static NSString * const _ERASURE = @"erasure";
static NSString * const _CREATED = @"created";
static NSString * const _DECRYPTED = @"isDecrypted";
static NSString * const _SIZE = @"size";
static NSString * const _STARRED = @"isStarred";
static NSString * const _SYNCED = @"isSynced";
static NSString * const _DOWNLOAD_STATE = @"downloadState";
static NSString * const _FILE_HANDLE = @"fileHandle";
static NSString * const _FILE_URI = @"fileUri";
static NSString * const _FILE_THUMBNAIL = @"thumbnail";

static NSString * const _FILE_FK = @"bucketId";

+(NSString *) createTable {

  return [NSString stringWithFormat: @"create table if not exists %@ (\
%@ TEXT primary key not null, \
%@ TEXT not null, \
%@ TEXT, \
%@ TEXT, \
%@ TEXT, \
%@ TEXT, \
%@ TEXT not null, \
%@ INTEGER, \
%@ INTEGER, \
%@ INTEGER, \
%@ INTEGER, \
%@ INTEGER DEFAULT 0, \
%@ INTEGER DEFAULT 0, \
%@ TEXT, \
%@ TEXT not null, \
%@ TEXT, \
FOREIGN KEY(%@) REFERENCES buckets(%@) ON DELETE CASCADE);", _TABLE_NAME, _FILE_ID, _NAME,
          _MIME_TYPE, _INDEX, _HMAC, _ERASURE, _CREATED, _DECRYPTED, _STARRED, _SIZE, _SYNCED,
          _DOWNLOAD_STATE, _FILE_HANDLE, _FILE_URI, _FILE_FK, _FILE_THUMBNAIL, _FILE_FK, _FILE_ID];
}

+(NSString *) TABLE_NAME {

  return _TABLE_NAME;
}

//+(NSString *) ID{
//  return _ID;
//}

+(NSString *) FILE_ID {

  return _FILE_ID;
}

+(NSString *) NAME {

  return _NAME;
}

+(NSString *) MIME_TYPE {

  return _MIME_TYPE;
}

+(NSString *) INDEX {

  return _INDEX;
}

+(NSString *) HMAC {

  return _HMAC;
}

+(NSString *) ERASURE {

  return _ERASURE;
}

+(NSString *) CREATED {

  return _CREATED;
}

+(NSString *) DECRYPTED {

  return _DECRYPTED;
}

+(NSString *) SIZE {

  return _SIZE;
}

+(NSString *) STARRED {

  return _STARRED;
}

+(NSString *) SYNCED {

  return _SYNCED;
}

+(NSString *) DOWNLOAD_STATE {

  return _DOWNLOAD_STATE;
}

+(NSString *) FILE_HANDLE {

  return _FILE_HANDLE;
}

+(NSString *) FILE_URI {

  return _FILE_URI;
}

+(NSString *) FILE_THUMBNAIL {

  return _FILE_THUMBNAIL;
}

+(NSString *) FILE_FK {

  return _FILE_FK;
}

@end

File: android/app/src/main/java/io/storj/mobile/storjlibmodule/dataprovider/contracts/FileContract.java
package io.storj.mobile.storjlibmodule.dataprovider.contracts;

/**
 * Created by Crawter on 02.03.2018.
 */

public final class FileContract extends BaseContract {
    public final static String TABLE_NAME = "files";

    public final static String _FILE_ID = "fileId";
    public final static String _NAME = "name";
    public final static String _MIMETYPE = "mimeType";
    public final static String _INDEX = "_index";
    public final static String _HMAC = "hmac";
    public final static String _ERASURE = "erasure";
    public final static String _CREATED = "created";
    public final static String _DECRYPTED = "isDecrypted";
    public final static String _SIZE = "size";
    public final static String _STARRED = "isStarred";
    public final static String _SYNCED = "isSynced";
    public final static String _DOWNLOAD_STATE = "downloadState";
    public final static String _FILE_HANDLE = "fileHandle";
    public final static String _FILE_URI = "fileUri";
    public final static String _FILE_THUMBNAIL = "thumbnail";

    public final static String FILE_FK = "bucketId";

    public static String createTable() {
        return String.format(
                "create table if not exists %s (" +
                "%s TEXT primary key not null, " +
                "%s TEXT not null, " +
                "%s TEXT, " +
                "%s TEXT, " +
                "%s TEXT, " +
                "%s TEXT, " +
                "%s TEXT not null, " +
                "%s INTEGER, " +
                "%s INTEGER, " +
                "%s INTEGER, " +
                "%s INTEGER, " +
                "%s INTEGER DEFAULT 0, " +
                "%s INTEGER DEFAULT 0, " +
                "%s TEXT, " +
                "%s TEXT not null, " +
                "%s TEXT, " +
                "FOREIGN KEY(%s) REFERENCES buckets(%s) ON DELETE CASCADE)",
                TABLE_NAME, _ID, _NAME, _MIMETYPE, _INDEX, _HMAC, _ERASURE, _CREATED, _DECRYPTED, _STARRED, _SIZE, _SYNCED, _DOWNLOAD_STATE, _FILE_HANDLE, _FILE_URI, FILE_FK, _FILE_THUMBNAIL, FILE_FK, _ID );
    }
}
